File: js/sumOfFoundIndexes.js
// Problem# 178
// Create a function which takes in an array of numbers and a number to find. Return the sum of every index in the
// array which matches the chosen number.
// https://edabit.com/challenge/TTu48ECaTRwZDj69x

const sumFoundIndexes = (arr, n) =>
  arr.reduce((acc, curr, i) => {
    if (curr === n) {
      acc += i;
    }

    return acc;
  }, 0);

module.exports = sumFoundIndexes;

File: python/sum_of_found_indexes.py
"""
Sum of Found Indexes

Create a function which takes in a list of numbers and a number to find. Return the sum of every index in the list which matches the chosen number.

https://edabit.com/challenge/YKgzPFZKt9XHXQHAu
"""
from typing import List


def sum_found_indexes(lst: List[int], n: int) -> int:
    return sum(i for i, value in enumerate(lst) if value == n)
